<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="DESKTOP-2KD9ONT" failures="4" tests="8" name="Surefire test" time="14.379" errors="0" timestamp="2022-10-11T22:15:42 IDT">
  <testcase classname="com.qacart.todo.testcases.UserTest" name="ShouldNotBeAbleToLoginIfPasswordIsNotCorrect" time="4.7"/>
  <testcase classname="com.qacart.todo.testcases.TodoTest" name="shouldBeAbleToAddTodo" time="5.011">
    <failure type="java.lang.RuntimeException" message="com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field &amp;quot;message&amp;quot; (class com.qacart.todo.models.Todo), not marked as ignorable (6 known properties: &amp;quot;isCompleted&amp;quot;, &amp;quot;item&amp;quot;, &amp;quot;userID&amp;quot;, &amp;quot;_id&amp;quot;, &amp;quot;createdAt&amp;quot;, &amp;quot;__v&amp;quot;])
 at [Source: (String)&amp;quot;{&amp;quot;message&amp;quot;:&amp;quot;Please Fill a correct Password&amp;quot;}&amp;quot;; line: 1, column: 13] (through reference chain: com.qacart.todo.models.Todo[&amp;quot;message&amp;quot;])">
      <![CDATA[java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "message" (class com.qacart.todo.models.Todo), not marked as ignorable (6 known properties: "isCompleted", "item", "userID", "_id", "createdAt", "__v"])
 at [Source: (String)"{"message":"Please Fill a correct Password"}"; line: 1, column: 13] (through reference chain: com.qacart.todo.models.Todo["message"])
at io.restassured.internal.path.json.mapping.JsonPathJackson2ObjectDeserializer.deserialize(JsonPathJackson2ObjectDeserializer.java:30)
at io.restassured.path.json.mapping.JsonPathObjectDeserializer$deserialize.call(Unknown Source)
at io.restassured.internal.mapping.Jackson2Mapper.deserialize(Jackson2Mapper.groovy:58)
at io.restassured.mapper.ObjectMapper$deserialize$1.call(Unknown Source)
at io.restassured.internal.mapping.ObjectMapping.parseWithJackson2(ObjectMapping.groovy:254)
at io.restassured.internal.mapping.ObjectMapping$parseWithJackson2$2.callStatic(Unknown Source)
at io.restassured.internal.mapping.ObjectMapping.deserialize(ObjectMapping.groovy:56)
at io.restassured.internal.mapping.ObjectMapping$deserialize$1.call(Unknown Source)
at io.restassured.internal.RestAssuredResponseOptionsGroovyImpl.as(RestAssuredResponseOptionsGroovyImpl.groovy:231)
at io.restassured.internal.RestAssuredResponseOptionsImpl.as(RestAssuredResponseOptionsImpl.java:169)
at com.qacart.todo.testcases.TodoTest.shouldBeAbleToAddTodo(TodoTest.java:27)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "message" (class com.qacart.todo.models.Todo), not marked as ignorable (6 known properties: "isCompleted", "item", "userID", "_id", "createdAt", "__v"])
 at [Source: (String)"{"message":"Please Fill a correct Password"}"; line: 1, column: 13] (through reference chain: com.qacart.todo.models.Todo["message"])
at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1127)
at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2023)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1700)
at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1678)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:320)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674)
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629)
at io.restassured.internal.path.json.mapping.JsonPathJackson2ObjectDeserializer.deserialize(JsonPathJackson2ObjectDeserializer.java:28)
... 25 more
... Removed 12 stack frames]]>
    </failure>
  </testcase> <!-- shouldBeAbleToAddTodo -->
  <testcase classname="com.qacart.todo.testcases.UserTest" name="shouldBeAbleToLogin" time="1.427"/>
  <testcase classname="com.qacart.todo.testcases.UserTest" name="shouldBeAbleToRegister" time="0.883"/>
  <testcase classname="com.qacart.todo.testcases.TodoTest" name="shouldBeAbleToDeleteTodoByID" time="2.643">
    <failure type="java.lang.IllegalStateException" message="Cannot parse object because no supported Content-Type was specified in response. Content-Type was &amp;apos;text/html; charset=utf-8&amp;apos;.">
      <![CDATA[java.lang.IllegalStateException: Cannot parse object because no supported Content-Type was specified in response. Content-Type was 'text/html; charset=utf-8'.
at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:58)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
at io.restassured.internal.mapping.ObjectMapping.deserialize(ObjectMapping.groovy:95)
at io.restassured.internal.mapping.ObjectMapping$deserialize$1.call(Unknown Source)
at io.restassured.internal.RestAssuredResponseOptionsGroovyImpl.as(RestAssuredResponseOptionsGroovyImpl.groovy:231)
at io.restassured.internal.RestAssuredResponseOptionsImpl.as(RestAssuredResponseOptionsImpl.java:169)
at com.qacart.todo.testcases.TodoTest.shouldBeAbleToDeleteTodoByID(TodoTest.java:66)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
... Removed 17 stack frames]]>
    </failure>
  </testcase> <!-- shouldBeAbleToDeleteTodoByID -->
  <testcase classname="com.qacart.todo.testcases.UserTest" name="shouldNotBeAbleToRegisterWithTheSameEmail" time="1.593"/>
  <testcase classname="com.qacart.todo.testcases.TodoTest" name="shouldBeAbleToGetATodoByID" time="2.139">
    <failure type="java.lang.IllegalStateException" message="Cannot parse object because no supported Content-Type was specified in response. Content-Type was &amp;apos;text/html; charset=UTF-8&amp;apos;.">
      <![CDATA[java.lang.IllegalStateException: Cannot parse object because no supported Content-Type was specified in response. Content-Type was 'text/html; charset=UTF-8'.
at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:108)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
at io.restassured.internal.mapping.ObjectMapping.deserialize(ObjectMapping.groovy:95)
at io.restassured.internal.mapping.ObjectMapping$deserialize$1.call(Unknown Source)
at io.restassured.internal.RestAssuredResponseOptionsGroovyImpl.as(RestAssuredResponseOptionsGroovyImpl.groovy:231)
at io.restassured.internal.RestAssuredResponseOptionsImpl.as(RestAssuredResponseOptionsImpl.java:169)
at com.qacart.todo.testcases.TodoTest.shouldBeAbleToGetATodoByID(TodoTest.java:53)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
... Removed 17 stack frames]]>
    </failure>
  </testcase> <!-- shouldBeAbleToGetATodoByID -->
  <testcase classname="com.qacart.todo.testcases.TodoTest" name="shouldNotBeAbleToAddTodoIfIsCompletedMissing" time="4.502">
    <failure type="java.lang.AssertionError" message="
Expected: is &amp;quot;The email and password combination is not correct, please fill a correct email and password&amp;quot;
     but: was &amp;quot;Please Fill a correct Password&amp;quot;">
      <![CDATA[java.lang.AssertionError: 
Expected: is "The email and password combination is not correct, please fill a correct email and password"
     but: was "Please Fill a correct Password"
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
at com.qacart.todo.testcases.TodoTest.shouldNotBeAbleToAddTodoIfIsCompletedMissing(TodoTest.java:44)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
... Removed 12 stack frames]]>
    </failure>
  </testcase> <!-- shouldNotBeAbleToAddTodoIfIsCompletedMissing -->
</testsuite> <!-- Surefire test -->
